app/resources/api/v2/tube_rack_resource.rb

```diff
@@ -14,33 +14,29 @@ module V2
     # or look at the [JSONAPI::Resources](http://jsonapi-resources.com/) package for Sequencescape's implementation
     # of the JSON:API standard.
     class TubeRackResource < BaseResource
-      # Constants...
-
       # TODO: Here be dragons! This resource is mutable and can be created via
       #       the JSON API. However the asset_creation record is not generated
       #       as we would be relying on the request to tell us who requested it.
       #       Instead this should be done as part of adding authentication to
       #       the API in the security OKR.
 
-      # model_name / model_hint if required
-
       default_includes :uuid_object, :barcodes
 
-      # Associations:
-      has_many :racked_tubes
-      has_many :comments, readonly: true
-      has_one :purpose, foreign_key: :plate_purpose_id
-
       # Attributes
-      attribute :uuid, readonly: true
       attribute :created_at, readonly: true
-      attribute :updated_at, readonly: true
       attribute :labware_barcode, write_once: true
-      attribute :size
-      attribute :number_of_rows, write_once: true
-      attribute :number_of_columns, write_once: true
       attribute :name, write_once: true
-      attribute :tube_locations
+      attribute :number_of_columns, write_once: true
+      attribute :number_of_rows, write_once: true
+      attribute :size
+      attribute :tube_locations, writeonly: true
+      attribute :updated_at, readonly: true
+      attribute :uuid, readonly: true
```

Comment:
@andrewsparkes I don't know if any of the changes I've made here are going to conflict with what you're doing.  Sorry if they do.  The newer test file will hopefully help to explain what to expect from one of these resources, hence the changes.